### DISCUSSION TITLE
What is the difference between REPL "in your project" and "not in your project"

### DISCUSSION THREAD

**Discussion by rafd** [BetterThanTomorrow/calva#1072] [Q&A]
Calva gives the following options when connecting to a REPL:
<img width="603" alt="Screen Shot 2021-03-18 at 9 29 44 AM" src="https://user-images.githubusercontent.com/89664/111634079-8a66c080-87cc-11eb-8c51-9fdbf9e2502b.png">

I understand that (1) is "have Calva start the REPL for you", but, I'm confused as to the difference between (2) and (3), ie. what is the difference between "...REPL in your project" and "...REPL not in your project"?

**Comment by PEZ** [comment]
Hello! So here you are @rafd , the Rafal Dittwald that made [this talk](https://www.youtube.com/results?search_query=solving+problems+the+clojure+way), right? Sorry for going off-topic right off the bat, but I must say that that talk is right up there among the best talks I have ever seen.

On-topic then. Those options differ in that one will look for `.nrepl-port` and project configuration files in order to let you select some different options. While the one *not in your project* will skip that and just ask for the `host:port` to connect to. So, maybe the wording is wrong there. It could still be **your** project. A use case: I often have a `babashka` nrepl server running on my machine and I use the last ”non-project” to connect to it now and then. Another use case could be that you want to connect to a repl on some machine somewhere for whatever reasons.

**Reply by rafd** [reply]
Yep, that's me. Thank you for the high praise! I was pleasantly surprised at how well it was received.
Since we're in a thankful mood, I'd also like to give you thanks for your work on Calva; although I don't use it as my day-to-day, I do recommend it to beginner Clojurians, and I think it's existence has made a big impact on helping get more people trying Clojure.

Re: the UI, with a running nrepl (through lein), the experience seems to be identical given either choice.

If you're able to auto-detect the various REPL files, perhaps a good UX could be to show the following options:
- Start a new REPL 
- Connect to running REPL (nREPL 1234)    (ie. autodetected)
- Connect to running REPL (babashka 3456)   (ie. autodetected)
- Connect to a different running REPL
- Start the "Getting Started" REPL

If you like that proposal, I can do the PR for it.

**Reply by PEZ** [reply]
Excellent suggestions. Please file an issue about it. Some things to consider:

I don't think there is a way to autodetect a babashka REPL.

There are three ”Start a REPL” options:
1. Start a REPL for the project (aka Jack-in)
2. Start a standalone REPL
3. Start the ”Getting Started REPL”

The second and third are implemented the same way, but targets different use cases.
